modules/core/shared/src/main/scala/data/LTree.scala

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2018-2021 by Rob Norris
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package skunk.data
+
+import cats.Eq
+
+final case class LTree private (labels: List[String]) {
+
+  def isAncestorOf(other: LTree): Boolean = {
+    labels.isEmpty ||                 // Empty LTree is parent to all
+    other.labels.startsWith(labels)   // Other labels starts with this labels
+  }
+
+  def isDescendantOf(other: LTree): Boolean = other.isAncestorOf(this)
+  
+  override def toString: String = labels.mkString(LTree.Separator.toString())
+}
+
+object LTree {
+  val Empty = LTree(Nil)
+
+  def unsafe(labels: String*): LTree = LTree(labels.toList)
```

Comment:
Let's leave this out. The user can deal with the case where they're sure it's valid. We could also add a `ltree"..."` literal handler at some point.

Reply:
I can try to add a literal. I've never built one before but seems like the `Literally` library might be a good way to go? Or would you prefer to avoid adding the dependency? At a quick glance it looks like it would need to be done independently for scala 2/3.

Reply:
Yes, Literally is the way to go. We can save it for a followup.